test/controllers/declaration/ExpressConsignmentControllerSpec.scala

```diff
@@ -0,0 +1,162 @@
+import play.api.test.Helpers._
+import play.twirl.api.HtmlFormat
+import views.html.declaration.express_consignment
+
+class ExpressConsignmentControllerSpec extends ControllerSpec {
+
+  private val expressConsignmentPage = mock[express_consignment]
+
+  val controller = new ExpressConsignmentController(
+    mockAuthAction,
+    mockJourneyAction,
+    mockExportsCacheService,
+    navigator,
+    stubMessagesControllerComponents(),
+    expressConsignmentPage
+  )(ec)
+
+  override protected def beforeEach(): Unit = {
+    super.beforeEach()
+    authorizedUser()
+    withNewCaching(aDeclaration(withType(DeclarationType.STANDARD)))
+    when(expressConsignmentPage.apply(any[Mode], any[Form[YesNoAnswer]])(any(), any())).thenReturn(HtmlFormat.empty)
+  }
+
+  override protected def afterEach(): Unit = {
+    reset(expressConsignmentPage)
+    super.afterEach()
+  }
+
+  def theResponseForm: Form[YesNoAnswer] = {
+    val captor = ArgumentCaptor.forClass(classOf[Form[YesNoAnswer]])
+    verify(expressConsignmentPage).apply(any(), captor.capture())(any(), any())
+    captor.getValue
+  }
+
+  override def getFormForDisplayRequest(request: Request[AnyContentAsEmpty.type]): Form[_] = {
+    await(controller.displayPage(Mode.Normal)(request))
+    theResponseForm
+  }
+
+  "ExpressConsignmentController" should {
+
+    onJourney(STANDARD, CLEARANCE, SIMPLIFIED, OCCASIONAL) { implicit request =>
+      "return 200 (OK)" when {
+
+        "display page method is invoked and cache is empty" in {
+          val result = controller.displayPage(Mode.Normal)(getRequest())
+          status(result) must be(OK)
+          verifyPageInvoked
+        }
+
+        "display page method is invoked and cache is not empty" in {
+          val payment = Some(TransportPayment(cash))
+          withNewCaching(aDeclaration(withTransportPayment(payment)))
+
+          val result = controller.displayPage(Mode.Normal)(getRequest())
+          status(result) must be(OK)
+          verifyPageInvoked
+        }
+      }
+
+      "return 303 (SEE_OTHER) and redirect to the 'Transport payment method' page" when {
+        "answer is 'yes'" in {
+          verifyRedirect(Some(YesNoAnswers.yes), Some(routes.TransportPaymentController.displayPage()))
+        }
+      }
+
+      "return 303 (SEE_OTHER) and redirect to 'Transport container' page" when {
+        "answer is 'no'" in {
+          verifyRedirect(Some(YesNoAnswers.no), Some(routes.TransportContainerController.displayContainerSummary()))
+        }
+      }
+
+      "return 400 (BAD_REQUEST)" when {
+
+        "form contains incorrect values" in {
+          val incorrectForm = Json.obj("yesNo" -> "wrong")
+
+          val result = controller.submitForm(Mode.Normal)(postRequest(incorrectForm))
+          status(result) must be(BAD_REQUEST)
+          verifyPageInvoked
+        }
+
+        "neither Yes or No have been selected on the page" in {
+          val incorrectForm = Json.obj("yesNo" -> "")
```

Comment:
Nit: isn't this the same test as above (incorrect value)? If the form is not selected should you not pass an empty json object (i.e. not even have the "yesNo" key either)?

Reply:
:+1: